stellargraph/mapper/sequences.py

```diff
@@ -527,3 +530,128 @@ def __len__(self):
 
     def __getitem__(self, index):
         return self.inputs, self.targets
+
+
+@experimental(reason="Missing unit tests.")
+class GraphSequence(Sequence):
+    """
+    A Keras-compatible data generator for training and evaluating graph classification models.
+    Use this class with the Keras methods :meth:`keras.Model.fit`,
+        :meth:`keras.Model.evaluate`, and
+        :meth:`keras.Model.predict`,
+
+    This class should be created using the `.flow(...)` method of
+    :class:`GraphGenerator`.
+
+    Args:
+        graphs (list)): The graphs as StellarGraph objects.
+        targets (np.ndarray, optional): An optional array of graph targets of size (N x C),
+            where N is the number of graphs and C is the target size (e.g., number of classes.)
+        normalize_adj (bool, optional): Specifies whether the adjacency matrix for each graph should
+            be normalized or not. The default is True.
+        batch_size (int, optional): The batch size. It defaults to 1.
+        name (str, optional): An optional name for this generator object.
+    """
+
+    def __init__(
+        self, graphs, targets=None, normalize_adj=True, batch_size=1, name=None
+    ):
+
+        self.name = name
+        self.graphs = graphs
+        self.normalize_adj = normalize_adj
+        self.targets = targets
+        self.batch_size = batch_size
+        # we assume that all graphs have node features of the same dimensionality
+        self.node_features_size = graphs[0].node_features(graphs[0].nodes()).shape[1]
+
+        if targets is not None:
+            if len(graphs) != len(targets):
+                raise ValueError(
+                    "When passed together targets and the number of graphs should be the same length."
+                )
+
+            self.targets = np.asanyarray(targets)
+
+        self.on_epoch_end()
+
+    def __len__(self):
+        return int(np.ceil(len(self.graphs) / self.batch_size))
+
+    def __normalize_adj(self, adj):
+        adj = adj.tolil()
+        adj.setdiag(1)  # add self loops
+        degree_matrix_diag = 1.0 / np.sqrt(adj.sum(axis=1))
+        degree_matrix_diag = np.squeeze(np.asarray(degree_matrix_diag))
+        degree_matrix = sparse.lil_matrix(adj.shape)
+        degree_matrix.setdiag(degree_matrix_diag)
+        degree_matrix = degree_matrix.tocsr()
+        adj = degree_matrix @ adj @ degree_matrix
+        return adj
+
+    def __getitem__(self, index):
+        graphs = self.graphs[
+            index * self.batch_size : (index * self.batch_size) + self.batch_size
+        ]
+        adj_graphs = [graph.to_adjacency_matrix() for graph in graphs]
```

Comment:
This has to recompute the adjacency matrices for each matrix in every epoch. What do you think about computing them once, earlier?

Reply:
The usual time vs space tradeoff. I think in this case, we should expect each graph to be small in the order of hundreds of nodes so pre-computing should be fine. Memory usage might start to become an issue if the number of graphs is very large which could happen if each is a subgraph of a very large graph, i.e., financial transaction network. 

I'll update it to pre-compute for now. We can come up with a heuristic later on to determine if pre-computing might use too much memory and not do it. I think it might be best that we go for speed now since all benchmark datasets tend to me small.

Ultimately, we want this normalisation done in a layer but the normalisation layers we have at the moment are not suitable because of how they handle the batch dimension.